executor/api/rest/server.go

```diff
@@ -139,6 +150,24 @@ func (r *SeldonRestApi) Initialise() {
 	}
 }
 
+type CloudeventHeaderMiddleware struct {
+	deploymentName string
+}
+
+func (h *CloudeventHeaderMiddleware) Middleware(next http.Handler) http.Handler {
+	// TODO: Write a test that ensures the initialise function always sets deploymentName
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		puid := r.Header.Get(payload.SeldonPUIDHeader)
+		w.Header().Set(CLOUDEVENTS_HEADER_ID_NAME, puid)
+		w.Header().Set(CLOUDEVENTS_HEADER_SPECVERSION_NAME, CLOUDEVENTS_HEADER_SPECVERSION_DEFAULT)
+		w.Header().Set(CLOUDEVENTS_HEADER_PATH_NAME, r.URL.Path)
+		w.Header().Set(CLOUDEVENTS_HEADER_TYPE_NAME, "seldon."+h.deploymentName+".response")
```

Comment:
Would it make sense to define a KNative `EventType` to make these more visible? Although this probably falls outside of the scope of this PR.

Reply:
Yeah makes sense, probably as a follow-up as this will be the first attempt to support KNative events so would be good to see where this goes from here